src/App.js

```diff
@@ -117,6 +118,8 @@ const AdScreener = (params) => {
   const [showNeedLoginModal, setShowNeedLoginModal] = useState(false);
   const handleShowNeedLoginModal = () => setShowNeedLoginModal(true);
   const handleCloseNeedLoginModal = () => setShowNeedLoginModal(false);
+  const [isAdDataLoaded, setIsAdDataLoaded] = useState(true);
```

Comment:
yes that makes sense.

Reply:
If I change this to isGetAdsRequestPending, the variable name doesn't make sense in this line:
      .then((response) => {
        console.log(response.data);
        setAds(response.data);
        if(response.data.length !== 0){
          setIsAdDataEmpty(true);
        }
        setIsGetAdsRequestPending(true); // It is actually not pending. 
I've set the initial state as true to prevent the loading icon from displaying before the getAds button is clicked, so it doesn't seem to make sense. But if I change the variable name, it doesn't make sense in the actual logic.

Reply:
sorry for not being clear. I mean changing the values as well. Right now I'm thinking about the idea of the variable, ie are we tracking if ad data has been loaded or tracking if a request is pending.

here's what I had in mind https://github.com/OnlinePoliticalTransparency/ad_screener/pull/28/files

Reply:
Ohh okay I get it ! The naming makes much more sense in this way.